[Docs] Add team deployment in existing machine and `detach_run` in docs (#4913)
skypilot-org/skypilot: docs/source/reservations/existing-machines.rst

   .. tip::

>       This is good for your personal usage. To grant teammates access to these machines, SkyPilot you can further deploy a :ref:`SkyPilot API server <sky-api-server>` and

Reviewer: Wdym? Not sure I follow the transition here.

   .. tip::

>       To grant teammates access to these machines through SkyPilot, you can further deploy a :ref:`SkyPilot API server <sky-api-server>` and
>       share the endpoint with them.

Reviewer: ```suggestion
      To enable shared access to a Kubernetes cluster, you can deploy a :ref:`SkyPilot API server <sky-api-server>`.
```
